### PR TITLE
Sniffer module

### PR DESCRIPTION
**Pull request by carloosamarante**

### CODE REVIEW

File: nosepass/domain_list.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from Sublist3r.sublist3r import main
-from concurrent.futures import ThreadPoolExecutor
-from concurrent.futures._base import TimeoutError
+from concurrent.futures import ThreadPoolExecutor, TimeoutError
 from nosepass.custom_json_logger import getLogger
+from numpy import concatenate
```
Comment (resilva87) [inline nosepass/domain_list.py]: @carloosamarante do you really need numpy for this? :eyes:

File: Pipfile [hunk 2]
```diff
@@ -21,6 +21,7 @@ python-nmap = "==0.6.1"
 pysocks = "==1.7.0"
 packaging = "==19.1"
 marshmallow = "==3.2.0"
+numpy = "==1.17.2"
```
Comment (shikow) [inline Pipfile]: :wat:

File: nosepass/sniffer.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from nosepass.domain_list import SubdomainList
+from nosepass.attack_surface_discoverer import discover
+
+
+def sniff(target_domains):
+    subdomains = SubdomainList().list_each_domain_subdomains(target_domains)
+
+    attack_surfaces = []
```
Comment (resilva87) [inline nosepass/sniffer.py]: @carloosamarante you can refactor this as `attack_surfaces = [discover(subdomain) for subdomain in subdomains]` if you like

File: tests/test_http_requestor.py [hunk 4]
```diff
@@ -40,3 +40,9 @@ def test_that_should_not_raises_connection_error(self, get):
         response = do_get('www.google.com')
 
         self.assertIsNotNone(response)
+
+    @patch.object(Session, 'get', side_effect=ReadTimeout)
+    def test_that_should_not_raises_timeout_error(self, get):
```
Comment (resilva87) [inline tests/test_http_requestor.py]: @carloosamarante you mean `should_raises`, don't you?

File: nosepass/sniffer.py (hunk 3 above)
Comment (rasecoiac03) [reply nosepass/sniffer.py]: or just `return [discover(subdomain) for subdomain in subdomains]`

File: nosepass/domain_list.py (hunk 1 above)
Comment (julianoborba) [reply nosepass/domain_list.py]: Removing

File: Pipfile (hunk 2 above)
Comment (julianoborba) [reply Pipfile]: Removed

File: tests/test_http_requestor.py (hunk 4 above)
Comment (julianoborba) [reply tests/test_http_requestor.py]: @resilva87 in this case, I was expecting that the function do_get shouldn't raises the exception, although the session.get could. Do that description looks weird?

File: nosepass/sniffer.py (hunk 3 above)
Comment (julianoborba) [reply nosepass/sniffer.py]: Done

**Comment by julianoborba** [comment]
Hey guys, can you please check again?

File: tests/test_http_requestor.py (hunk 4 above)
Comment (resilva87) [reply tests/test_http_requestor.py]: @julianoborba kind of, I didn't understand from the description :trollface:

Comment (julianoborba) [reply tests/test_http_requestor.py]: I will adjust that

**Pull request merged**